### PR TITLE
Use 'dev' sentry environment for airbyte-ci pipx installs in PRs

### LINKED ISSUE #33892: airbyte-ci: don't report sentry issues on dev branches
Our #connectors-ci-errors gets loads of errors whenever we are developing a new feature or changing things on airbyte-ci. We are able to recognize when we're developing - we could mess with the SENTRY_DSN environment variable or create a new one specifically for sentry ignoring. 

I believe we could mess with the value [here](https://github.com/airbytehq/airbyte/pull/33519/files#diff-8bba74a834b43cc4f45b74d9e25f898d291bc2d8c537e423325e4da1487efaa1R159), only setting it in the case of the binary install 

Pros: 
* #connectors-ci-errors becomes less noisy and more actionable

Cons: 
* sentry no longer pings on all changes to airbyte ci, including e.g. small changes made by users outside of our team. that could mean we miss things

### PR DESCRIPTION
**Pull request by erohmensing**
Use different sentry environments based on how we install Airbyte CI in PRs so that we can configure our alerting and sort our open issues on sentry. 

After changing our slack alerts (still not sure why we have 2 different ones) explicitly to alert for the `production` environment (which is what all of our issues are being sent to currently), a [new dev alert](https://airbytehq.sentry.io/issues/?environment=dev&project=4505596642983936&referrer=sidebar&statsPeriod=14d) was not posted to slack. 

Closes #33892

### CODE REVIEW

**Comment by erohmensing** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #33920** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33920?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/33920?utm_source=stack-comment).

**Review by evantahler — APPROVED** [review]
🥳

File: .github/actions/run-dagger-pipeline/action.yml [hunk 1]
```diff
@@ -101,6 +101,17 @@ runs:
           pipelines:
             - 'airbyte-ci/connectors/pipelines/**'
 
+    - name: Determine how Airbyte CI should be installed
```
Comment (bnchrch) [inline .github/actions/run-dagger-pipeline/action.yml:104]: Love this addition 😍

**Review by bnchrch — APPROVED** [review]
Just beautiful.

**Pull request merged**